### DISCUSSION TITLE
.env.production

### DISCUSSION THREAD

**Discussion by cfarvidson** [basecamp/kamal#288] [Ideas]
I think that the behaviour right now is that `.env` supersedes `.env.production` if that is present when doing a deploy.

I would like to have the possibility to use a switch to make mrsk choose `.env.production`.

Or just make it pick `.env.production` if present.

**Comment by tannakartikey** [comment]
I think the `dotenv` gem would consider the current environment. So what happens if you fire `RAILS_ENV=production bin/mrsk deploy`?

**Comment by cfarvidson** [comment]
When I try to run the `RAILS_ENV=production ` the following error appears.

	mrsk details
  	ERROR (KeyError): key not found: "NEXTAUTH_SECRET"

Basically saying that it can't find my NEXTAUTH_SECRET that's in the .env.production

If I create a `config/deploy.production.yml` and copy the `.env` -> `.env.production` and run `mrsk details` I get this really strange error:

	ERROR (SSHKit::Command::Failed): docker exit status: 256
	docker stdout: Nothing written
	docker stderr: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon 	running?

And jup. The docker service is running. And everything works great if I use `.env` and `config/deploy.yml`

**Comment by cfarvidson** [comment]
I think that this t just be an issue. Found https://github.com/mrsked/mrsk/issues/233
